src/main/java/uk/gov/hmcts/reform/em/stitching/template/TemplateRenditionClient.java

```diff
@@ -0,0 +1,69 @@
+package uk.gov.hmcts.reform.em.stitching.template;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import okhttp3.*;
+import org.apache.tomcat.util.http.fileupload.IOUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.em.stitching.service.impl.DocumentTaskProcessingException;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.UUID;
+
+@Component
+public class TemplateRenditionClient {
+
+    @Value("${docmosis.render.endpoint}")
+    private String docmosisRenderEndpoint;
+    @Value("${docmosis.accessKey}")
+    private String docmosisAccessKey;
+    private final OkHttpClient client;
+
+    @Autowired
+    public TemplateRenditionClient(@Autowired OkHttpClient client) {
+        this.client = client;
+    }
+
+    public File renderTemplate(String templateId, JsonNode payload) throws IOException, DocumentTaskProcessingException {
+        String tempFileName = String.format("%s%s",
```

Comment:
Why not use File.createTempFile here as well?

Reply:
as discussed with @linusnorton - this parameter is required by Docmosis API and same approach is taken in Doc Assembly